File: UF1/Practica 3 JavaDoc/elipse-workpage/src/Persona.java

/**
 * Esta es una clase principal que representa una persona.
 * @author Xiaobin Zhou
 */
public class Persona {
	private String nombre;
    private int edad;

    /**
     * Constructor de la clase Persona.
     * 
     * @param nombre El nombre de la persona.
     * @param edad La edad de la persona.
     */
    public Persona(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
    }

    /**
     * Obtiene el nombre de la persona.
     * 
     * @return El nombre de la persona.
     */
    public String getNombre() {
        return nombre;
    }
    
    /**
     * Establece el nombre de la persona.
     * 
     * @param nombre El nombre a establecer.
     */
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    /**
     * Obtiene la edad de la persona.
     * 
     * @return La edad de la persona.
     */
    public int getEdad() {
        return edad;
    }

    /**
     * Establece la edad de la persona.
     * 
     * @param edad La edad a establecer.
     */
    public void setEdad(int edad) {
        this.edad = edad;
    }

    /**
     * Saluda a la persona.
     * 
     * @return Un saludo que incluye el nombre de la persona.
     */
    public String saludar() {
        return "¡Hola! Mi nombre es " + nombre;
    }
    
    /**
     * Incrementa la edad de la persona en 1 y muestra un mensaje de cumpleaños.
     */
    public void birthday() {
        this.edad++;
        System.out.println("Feliz cumpleaños, ahora tengo " + this.edad + " años.");
    }

    /**
     * Revisa si la edad de la persona es válida.
     * 
     * @throws Exception Si la edad es un número negativo.
     */
    public void revisarEdad() throws Exception {
        if (edad < 0) {
            throw new Exception("La edad no puede ser un número negativo.");
        } else {
            System.out.println("La edad es válida.");
        }
    }
}

File: UF1/Practica 4 Doc/persona.py
class Persona:
    """
    Esta es una clase principal que representa una persona.
    """

    def __init__(self, nombre, edad):
        """
        Constructor de la clase Persona.

        :param nombre: El nombre de la persona.
        :param edad: La edad de la persona.
        """
        self.nombre = nombre
        self.edad = edad

    def get_nombre(self):
        """
        Obtiene el nombre de la persona.

        :return: El nombre de la persona.
        """
        return self.nombre

    def set_nombre(self, nombre):
        """
        Establece el nombre de la persona.

        :param nombre: El nombre a establecer.
        """
        self.nombre = nombre

    def get_edad(self):
        """
        Obtiene la edad de la persona.

        :return: La edad de la persona.
        """
        return self.edad

    def set_edad(self, edad):
        """
        Establece la edad de la persona.

        :param edad: La edad a establecer.
        """
        self.edad = edad

    def saludar(self):
        """
        Saluda a la persona.

        :return: Un saludo que incluye el nombre de la persona.
        """
        return f"¡Hola! Mi nombre es {self.nombre}"

    def birthday(self):
        """
        Incrementa la edad de la persona en 1 y muestra un mensaje de cumpleaños.
        """
        self.edad += 1
        print(f"Feliz cumpleaños, ahora tengo {self.edad} años.")

    def revisar_edad(self):
        """
        Revisa si la edad de la persona es válida.
        """
        if self.edad < 0:
            raise Exception("La edad no puede ser un número negativo.")
        else:
            print("La edad es válida.")